include/seqan3/alignment/matrix/matrix_concept.hpp

```diff
@@ -0,0 +1,141 @@
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \author Marcel Ehrhardt <marcel.ehrhardt AT fu-berlin.de>
+ * \brief Contains seqan3::detail::matrix_concept.
+ */
+
+#pragma once
+
+#include <cstddef>
+
+#include <seqan3/std/concepts>
+
+namespace seqan3::detail
+{
+
+/*!\interface seqan3::detail::matrix_concept <>
+ * \brief Defines the requirements of an matrix (e.g. score matrices, trace matrices).
+ * \tparam matrix_t The type the concept check is performed on (the putative matrix).
+ * \ingroup alignment_matrix
+ */
+/*!\name Requirements for seqan3::detail::matrix_concept
+* \brief You can expect these members on all types that implement seqan3::detail::matrix_concept.
+* \memberof seqan3::detail::matrix_concept
+* \{
+*/
+//!\cond
+template <typename matrix_t>
+concept matrix_concept = requires(matrix_t m)
+{
+//!\endcond
+    /*!\typedef typedef IMPLEMENTATION_DEFINED entry_type;
+     * \brief The type of an entry in the matrix.
+     * \memberof seqan3::detail::matrix_concept
+     */
+    typename matrix_t::entry_type;
+
+    /*!\fn std::size_t cols() const noexcept;
+     * \brief The number of columns in the matrix.
+     * \memberof seqan3::detail::matrix_concept
+     */
+    { m.cols() } -> std::size_t;
+
+    /*!\fn std::size_t rows() const noexcept;
+     * \brief The number of rows in the matrix.
+     * \memberof seqan3::detail::matrix_concept
+     */
+    { m.rows() } -> std::size_t;
+
+    /*!\fn entry_type at(unsigned row, unsigned col) const noexcept;
+     * \brief The entry of the matrix at position (\a row, \a col), e.g. `matrix[row][col]`.
+     * \memberof seqan3::detail::matrix_concept
+     */
+    { m.at(0u, 0u) } -> typename matrix_t::entry_type;
+//!\cond
+};
+//!\endcond
+//!\}
+
+/*!\brief `true`, if two alignment matrices are equal.
```

Comment:
This is not a full sentence. Also please group the comparison operator into named group called "Comparison operators".